lib/constructs/callback-interface.js

```diff
@@ -0,0 +1,232 @@
+"use strict";
+
+const utils = require("../utils.js");
+const Types = require("../types.js");
+const Constant = require("./constant.js");
+
+class CallbackInterface {
+  constructor(ctx, idl) {
+    this.ctx = ctx;
+    this.idl = idl;
+    this.name = idl.name;
+    this.str = null;
+
+    this.requires = new utils.RequiresMap(ctx);
+
+    this.operation = null;
+    this.constants = new Map();
+
+    this._analyzed = false;
+    this._outputStaticProperties = new Map();
+  }
+
+  _analyzeMembers() {
+    for (const member of this.idl.members) {
+      switch (member.type) {
+        case "operation":
+          if (this.operation !== null) {
+            throw new Error(
+              `Callback interface ${this.name} has more than one operation`
+            );
+          }
+          this.operation = member;
+          break;
+        case "const":
+          this.constants.set(member.name, new Constant(this.ctx, this, member));
+          break;
+        default:
+          if (!this.ctx.options.suppressErrors) {
```

Comment:
We probably don't need to guard this check. It's only used for things like unrecognized types.

Reply:
I copied this from `interface.js`, which has this check: <https://github.com/jsdom/webidl2js/blob/6b2784221206174e745e2756e7ed2f4f8d7bb75c/lib/constructs/interface.js#L301-L304>

Reply:
Heh. I think that might be because there are things allowed in interfaces that we don’t support yet, like async iterables. In contrast, anything else here would be in violation of the Web IDL spec.